src/arkode/arkode_lsrkstep.c

```diff
@@ -0,0 +1,2233 @@
+  }
+
+  /* Evaluate stages j = 2,...,step_mem->req_stages */
+  for (int j = 2; j <= 5; j++)
+  {
+#if SUNDIALS_LOGGING_LEVEL >= SUNDIALS_LOGGING_DEBUG
+    SUNLogger_QueueMsg(ARK_LOGGER, SUN_LOGLEVEL_DEBUG,
+                       "ARKODE::lsrkStep_TakeStepSSP104", "start-stage",
+                       "step = %li, stage = %i, h = %" RSYM ", tcur = %" RSYM,
+                       ark_mem->nst, j, ark_mem->h,
+                       ark_mem->tcur +
+                         ((sunrealtype)j - ONE) * onesixth * ark_mem->h);
+#endif
+
+    /* apply user-supplied stage postprocessing function (if supplied) */
+    if (ark_mem->ProcessStage != NULL)
+    {
+      retval = ark_mem->ProcessStage(ark_mem->tcur + ((sunrealtype)j - TWO) *
+                                                       onesixth * ark_mem->h,
+                                     ark_mem->ycur, ark_mem->user_data);
+      if (retval != 0) { return ARK_POSTPROCESS_STAGE_FAIL; }
+    }
+
+    retval = step_mem->fe(ark_mem->tcur +
+                            ((sunrealtype)j - ONE) * onesixth * ark_mem->h,
+                          ark_mem->ycur, ark_mem->fn, ark_mem->user_data);
+    step_mem->nfe++;
+    if (retval < 0) { return ARK_RHSFUNC_FAIL; }
+    if (retval > 0) { return RHSFUNC_RECVR; }
+
+#ifdef SUNDIALS_LOGGING_EXTRA_DEBUG
+    SUNLogger_QueueMsg(ARK_LOGGER, SUN_LOGLEVEL_DEBUG,
+                       "ARKODE::lsrkStep_TakeStepSSP104", "stage RHS",
+                       "F_%i(:) =", j);
+    N_VPrintFile(ark_mem->fn, ARK_LOGGER->debug_fp);
+#endif
+
+    N_VLinearSum(ONE, ark_mem->ycur, onesixth * ark_mem->h, ark_mem->fn,
+                 ark_mem->ycur);
+    if (j == 4 && !ark_mem->fixedstep)
+    {
+      N_VLinearSum(ONE, ark_mem->tempv1, SUN_RCONST(0.3) * ark_mem->h,
+                   ark_mem->fn, ark_mem->tempv1);
+    }
+  }
+  N_VLinearSum(SUN_RCONST(1.0) / SUN_RCONST(25.0), ark_mem->tempv2,
+               SUN_RCONST(9.0) / SUN_RCONST(25.0), ark_mem->ycur,
+               ark_mem->tempv2);
+  N_VLinearSum(SUN_RCONST(15.0), ark_mem->tempv2, SUN_RCONST(-5.0),
+               ark_mem->ycur, ark_mem->ycur);
+
+  /* apply user-supplied stage postprocessing function (if supplied) */
+  if (ark_mem->ProcessStage != NULL)
+  {
+    retval = ark_mem->ProcessStage(ark_mem->tcur + FOUR * onesixth * ark_mem->h,
+                                   ark_mem->ycur, ark_mem->user_data);
+    if (retval != 0) { return ARK_POSTPROCESS_STAGE_FAIL; }
+  }
+
+  for (int j = 6; j <= 9; j++)
+  {
+#if SUNDIALS_LOGGING_LEVEL >= SUNDIALS_LOGGING_DEBUG
+    SUNLogger_QueueMsg(ARK_LOGGER, SUN_LOGLEVEL_DEBUG,
+                       "ARKODE::lsrkStep_TakeStepSSP104", "start-stage",
+                       "step = %li, stage = %i, h = %" RSYM ", tcur = %" RSYM,
+                       ark_mem->nst, j, ark_mem->h,
+                       ark_mem->tcur +
+                         ((sunrealtype)j - FOUR) * onesixth * ark_mem->h);
+#endif
+
+    retval = step_mem->fe(ark_mem->tcur +
```

Comment:
Relocate postprocessing from above and update stage time
```suggestion
    /* apply user-supplied stage postprocessing function (if supplied) */
    if (ark_mem->ProcessStage != NULL)
    {
      retval = ark_mem->ProcessStage(ark_mem->tcur +
                                     ((sunrealtype)j - FOUR) * onesixth * ark_mem->h,
                                     ark_mem->ycur, ark_mem->user_data);
      if (retval != 0) { return ARK_POSTPROCESS_STAGE_FAIL; }
    }
    
    retval = step_mem->fe(ark_mem->tcur +
```